state/exchange/reducer.ts

```diff
@@ -0,0 +1,291 @@
+					const baseAmountNoFee = wei(state.baseBalance).mul(inverseRate);
+					const fee = baseAmountNoFee.mul(state.exchangeFeeRate ?? 0);
+					state.quoteAmount = truncateNumbers(baseAmountNoFee.add(fee), DEFAULT_CRYPTO_DECIMALS);
+				}
+			}
+		},
+		setOpenModal: (state, action) => {
+			state.openModal = action.payload;
+		},
+		closeModal: (state) => {
+			state.openModal = undefined;
+		},
+		setApprovalStatus: (state, action) => {
+			state.approvalStatus = action.payload;
+		},
+	},
+	extraReducers: (builder) => {
+		builder.addCase(fetchBalances.fulfilled, (state, action) => {
+			state.redeemableSynthBalances = action.payload.redeemableSynthBalances;
+			state.totalRedeemableBalance = action.payload.totalRedeemableBalance;
+		});
+		builder.addCase(fetchTransactionFee.fulfilled, (state, action) => {
+			state.transactionFee = action.payload.transactionFee;
+			state.feeCost = action.payload.feeCost;
+		});
+		builder.addCase(submitExchange.pending, (state) => {
+			state.isSubmitting = true;
+			state.openModal = 'confirm';
+			state.txError = undefined;
+		});
+		builder.addCase(submitExchange.fulfilled, (state) => {
+			state.isSubmitting = false;
+			state.openModal = undefined;
+		});
+		builder.addCase(submitExchange.rejected, (state, action) => {
+			state.isSubmitting = false;
+			state.openModal = undefined;
+			state.txError = action.error.message;
+		});
+		builder.addCase(submitApprove.pending, (state) => {
+			state.approvalStatus = 'approving';
+			state.openModal = 'approve';
+			state.txError = undefined;
+		});
+		builder.addCase(submitApprove.fulfilled, (state) => {
+			state.openModal = undefined;
+		});
+		builder.addCase(submitApprove.rejected, (state, action) => {
+			state.openModal = undefined;
+			state.txError = action.error.message;
+		});
+		builder.addCase(submitRedeem.pending, (state) => {
+			state.openModal = 'redeem';
+			state.txError = undefined;
+		});
+		builder.addCase(submitRedeem.fulfilled, (state) => {
+			state.openModal = undefined;
+		});
+		builder.addCase(submitRedeem.rejected, (state, action) => {
+			state.openModal = undefined;
+			state.txError = action.error.message;
+		});
+		builder.addCase(submitSettle.pending, (state) => {
+			state.openModal = 'settle';
+			state.txError = undefined;
+		});
+		builder.addCase(submitSettle.fulfilled, (state) => {
+			state.openModal = undefined;
+		});
+		builder.addCase(submitSettle.rejected, (state, action) => {
+			state.openModal = undefined;
+			state.txError = action.error.message;
+		});
+		builder.addCase(fetchTokenList.pending, (state) => {
+			state.tokenListLoading = true;
+		});
+		builder.addCase(fetchTokenList.fulfilled, (state, action) => {
+			state.tokenListLoading = false;
+			state.synthsMap = action.payload.synthsMap;
+			state.tokensMap = action.payload.tokensMap;
+			state.tokenList = action.payload.tokenList;
+		});
+		builder.addCase(fetchTokenList.rejected, (state) => {
+			state.tokenListLoading = false;
```

Comment:
It would be nice if we kept a consistent pattern for query states, like you done above on balances with `status: error | loading | success` I think we should stick to the pattern for all query states.